lib/ansible/utils/vars.py

```diff
@@ -165,33 +165,46 @@ def load_options_vars(version):
     return options_vars
 
 
-def isidentifier(ident):
+def isidentifier_PY3(ident):
     """
-    Determines, if string is valid Python identifier using the ast module.
+    Determines if string is valid Python identifier for Python 3.
     Originally posted at: http://stackoverflow.com/a/29586366
     """
 
     if not isinstance(ident, string_types):
         return False
 
-    try:
-        root = ast.parse(ident)
-    except SyntaxError:
+    if not ident.isidentifier():
         return False
 
-    if not isinstance(root, ast.Module):
+    if keyword.iskeyword(ident):
         return False
 
-    if len(root.body) != 1:
-        return False
+    return True
+
+
+def isidentifier_PY2(ident):
+    """
+    Determines if string is valid Python identifier for Python 2.
+    Originally posted at: http://stackoverflow.com/a/29586366
+    """
 
-    if not isinstance(root.body[0], ast.Expr):
+    if not isinstance(ident, string_types):
         return False
 
-    if not isinstance(root.body[0].value, ast.Name):
+    if not bool(_NAME_RE.match(ident)):
         return False
 
-    if root.body[0].value.id != ident:
+    if keyword.iskeyword(ident) or ident in ('True', 'False', 'None'):
         return False
 
     return True
+
+
+if PY3:
+    isidentifier = isidentifier_PY3
+else:
+    # https://python-future.org/_modules/future/utils.html#isidentifier
+    import re
+    _NAME_RE = re.compile(r"[a-zA-Z_][a-zA-Z0-9_]*$")
```

Comment:
This is not unicode frienly, see `INVALID_VARIABLE_NAMES` in `lib/ansible/constants.py` , we had to update to allow for unicode chars .. also you should be able to reuse

Reply:
+1